### ISSUE TITLE
How to train this on a custom dataset ?

### ISSUE THREAD

**Issue by dmenig** [wzmsltw/BSN-boundary-sensitive-network#35]
I've been looking at this for the past few days and I can't seem to figure out how I can use this to train on any other dataset than Thumos or ActivityNet. How would you train this in the end-to-end fashion proposed in the article please ?

**Comment by guancheng817** [comment]
same question..

**Comment by 95xueqian** [comment]
@hyperfraise  have you know how to train this model on other dataset? thanks

**Comment by dmenig** [comment]
I have, I also added a classification part, and made it really end-to-end. I'm not sure t's the most proper way of doing it but I'll show you soon.

**Comment by guancheng817** [comment]
> I have, I also added a classification part, and made it really end-to-end. I'm not sure t's the most proper way of doing it but I'll show you soon.

Thanks for your works, look forward to your open source..

**Comment by dmenig** [comment]
Here you have my work. I welcome criticism and error pointing !

https://github.com/hyperfraise/end-to-end-BSN.pytorch

**Comment by pengxiaoxiao** [comment]
Hello, I would like to ask what kind of input data you have implemented for end-to-end BSN. I don't understand some of your source code. I'm sorry!

**Comment by dmenig** [comment]
Videos of such shape (batch_size, channels, video_length, image_size, image_size)

**Comment by dmenig** [comment]
And labels of shape [class, t_start, t_end]

**Comment by dmenig** [comment]
See the https://github.com/hyperfraise/end-to-end-BSN.pytorch/blob/master/example.py file for demo on how to use it.

**Comment by pengxiaoxiao** [comment]
Is there any script file processed? Take the thmos2014 data set as an example.





------------------ 原始邮件 ------------------
发件人: "hyperfraise"<notifications@github.com>;
发送时间: 2019年7月18日(星期四) 晚上10:52
收件人: "wzmsltw/BSN-boundary-sensitive-network"<BSN-boundary-sensitive-network@noreply.github.com>;
抄送: "摸摸头^^"<2804597917@qq.com>;"Comment"<comment@noreply.github.com>;
主题: Re: [wzmsltw/BSN-boundary-sensitive-network] How to train this on acustom dataset ? (#35)




Videos of such shape (batch_size, channels, video_length, image_size, image_size)

**Comment by dmenig** [comment]
Custom datasets will have the particularity of having custom dataloader and custom formats of data storage, label storage etc... So I thought that the only thing I could actually do is tell the reader the format of the data that should be passed to the loss functions, and let the reader decide how (s)he needed to transform his(er) jsons and videos to this end.

However, you're right that it would allow for more flexibility on Thumos14 architectures. I'll try to implement a simple dataloader for this as soon as possible.

**Comment by pengxiaoxiao** [comment]
Thank you very much.

**Comment by pengxiaoxiao** [comment]
I would like to ask your model features or the features of brother tianwei. The video feature extraction was not added to the end-to-end architecture.



------------------ 原始邮件 ------------------
发件人: "hyperfraise"; 
发送时间: 2019年7月18日(星期四) 晚上10:53
收件人: "wzmsltw/BSN-boundary-sensitive-network";
抄送: "shawxiao"<2804597917@qq.com>;"Comment";
主题: Re: [wzmsltw/BSN-boundary-sensitive-network] How to train this on acustom dataset ? (#35)




And labels of shape [class, t_start, t_end]
 
&mdash;

**Comment by dmenig** [comment]
Hey. i'll respond to issues on my code, on the aforementionned repo. If you've got suggestions, I'll be glad to adress them there. I don't think here is the best place :)

https://github.com/hyperfraise/end-to-end-BSN.pytorch

**Comment by Malithi-gif** [comment]
@hyperfraise have you apply this into custom video dataset?

**Comment by dmenig** [comment]
Yes I failed to go above baseline accuracy. But I never could with temporal detection framework anyways :/

**Comment by Malithi-gif** [comment]
@hyperfraise, Thanks for your reply. What method use for get the video features? TSN?

**Comment by dmenig** [comment]
I trained it fully end to end with 2d resnet 18 feature extractor with the code I linked. So I extracted features on the fly.
